packages/fieldset/src/LionFieldset.js

```diff
@@ -222,7 +222,10 @@ export class LionFieldset extends FormRegistrarMixin(
     return serializedValues;
   }
 
-  resetGroup() {
+  async resetGroup() {
+    if (!this.submitted) {
```

Comment:
I'm not sure if this guard makes sense, or is even necessary. If no update is pending, `await this.updateComplete` should resolve immediately anyway.

So perhaps remove the guard and just add a comment: "To ensure resetting happens after submission process, in case it was fired right before reset"

Reply:
Fixed